src/lib/moment/start-end-of.js

```diff
@@ -16,8 +17,8 @@ export function startOf (units) {
         case 'isoWeek':
         case 'day':
         case 'date':
-            this.hours(0);
-            /* falls through */
+            setTime(this, 0, 0, 0, 0);
+            break;
         case 'hour':
             this.minutes(0);
```

Comment:
Seems like the `hour` case should be `this.setTime(this, this.hour(), 0, 0, 0)`, etc